wordpress-mobile/WordPress-Android
# Issue/12319 clear cards cache when necessary

Parent issue #12319 

Merge instructions
1. Review this PR
2. Check if https://github.com/wordpress-mobile/WordPress-Android/pull/12572 has been merged. If it was, update the target branch to develop.
5. Merge this PR

This PR takes care of clearing cached posts fetched from the /cards endpoint after we force fetch the first page. I don't like this solution, but I couldn't come up with anything better. We use a similar approach for "Bookmark/SavedForLater" posts. 

The solution introduces a new local only ReaderTag. All the posts fetched from the /cards endpoint are stored under this tag so it's easy to filter them and purge the cache after we fetch a new data.

To test:
1. Clear app data
1. Turn on the RI Feature Flag
2. Go to "Discover" tab and notice there are posts displayed
3. Go to following tab just to make sure you fetch some other posts
4. Open Stetho and check the first ten posts (ordered by rowId) - these are the posts for the discover tab
5. Kill the app and start it again
6. Open Discover
7. Open Stetho and notice the first ten posts are not there and they are now on the last 10 ids (they were removed and re-insterted)

PR submission checklist:
